django/db/migrations/executor.py

```diff
@@ -224,12 +224,15 @@ def _migrate_all_backwards(self, plan, full_plan, fake):
         # remaining applied migrations.
         last_unapplied_migration = plan[-1][0]
         state = states[last_unapplied_migration]
+        # Avoid mutating state with apps rendered as it's an expensive operation.
```

Comment:
If we want a better comment I'd need to dig into the state mutation to remember what specifically is expensive.